File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int PairSum(ListNode head)
    {
        var result = 0;
        var lst = GetList(head);
        var n = lst.Count;
        
        for (int i = 0; i < n/2; i++)
        {
            var twinVal = lst[n - 1 - i];
            
            var sum = twinVal + lst[i];
            result = Math.Max(sum, result);
        }
        
        return result;
    }
    public List<int> GetList(ListNode head)
    {
        var result = new List<int>();
        while (head != null)
        {
            result.Add(head.val);
            head = head.next;
        }
        return result;
    }
}

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def pairSum(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: int
        """
        next_node = head
        n = 0
        values = []
        while next_node != None:
            values.append(next_node.val)
            next_node = next_node.next
            n += 1
        greatest_twin_sum = 0
        for i, val in enumerate(values):
            twin = n - 1 - i
            if twin == 0:
                break
            twin_sum = values[i] + values[twin]
            if twin_sum > greatest_twin_sum:
                greatest_twin_sum = twin_sum
            
        return greatest_twin_sum
        